subsys/random/rand32_xoroshiro128.c

```diff
@@ -76,6 +78,32 @@ static int xoroshiro128_initialize(struct device *dev)
 
 	return 0;
 }
+#elif CONFIG_XOROSHIRO_DEV_ID_RANDOM_GENERATOR
+static u32_t xoroshiro128_next(void);
+
+static int xoroshiro128_initialize(struct device *dev)
+{
+	ssize_t id_len;
+	int i;
+
+	id_len = hwinfo_get_device_id((u8_t *)&state, sizeof(state));
+	if (id_len < 1) {
```

Comment:
Is the check here correct, `id_len < 1`?

Reply:
Negative number means the id is not available and zero would give no random numbers at all. With one we get at least somehow a random number.